api/v1alpha4/azureclustertemplate_types.go

```diff
@@ -0,0 +1,94 @@
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AzureClusterTemplateSpec defines the desired state of AzureClusterTemplate
+type AzureClusterTemplateSpec struct {
+	// NetworkSpec encapsulates all things related to Azure network.
+	NetworkSpec NetworkSpec `json:"networkSpec,omitempty"`
+
+	// +optional
+	ResourceGroup string `json:"resourceGroup,omitempty"`
+
+	// +optional
+	SubscriptionID string `json:"subscriptionID,omitempty"`
+
+	Location string `json:"location"`
+
+	// AdditionalTags is an optional set of tags to add to Azure resources managed by the Azure provider, in addition to the
+	// ones added by default.
+	// +optional
+	AdditionalTags Tags `json:"additionalTags,omitempty"`
+
+	// IdentityRef is a reference to an AzureIdentity to be used when reconciling this cluster
+	// +optional
+	IdentityRef *corev1.ObjectReference `json:"identityRef,omitempty"`
+
+	// AzureEnvironment is the name of the AzureCloud to be used.
+	// The default value that would be used by most users is "AzurePublicCloud", other values are:
+	// - ChinaCloud: "AzureChinaCloud"
+	// - GermanCloud: "AzureGermanCloud"
+	// - PublicCloud: "AzurePublicCloud"
+	// - USGovernmentCloud: "AzureUSGovernmentCloud"
+	// +optional
+	AzureEnvironment string `json:"azureEnvironment,omitempty"`
+
+	// BastionSpec encapsulates all things related to the Bastions in the cluster.
+	// +optional
+	BastionSpec BastionSpec `json:"bastionSpec,omitempty"`
+
+	// CloudProviderConfigOverrides is an optional set of configuration values that can be overridden in azure cloud provider config.
+	// This is only a subset of options that are available in azure cloud provider config.
+	// Some values for the cloud provider config are inferred from other parts of cluster api provider azure spec, and may not be available for overrides.
+	// See: https://kubernetes-sigs.github.io/cloud-provider-azure/install/configs
+	// Note: All cloud provider config values can be customized by creating the secret beforehand. CloudProviderConfigOverrides is only used when the secret is managed by the Azure Provider.
+	// +optional
+	CloudProviderConfigOverrides *CloudProviderConfigOverrides `json:"cloudProviderConfigOverrides,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Resource Group",type="string",priority=1,JSONPath=".spec.resourceGroup"
+// +kubebuilder:printcolumn:name="SubscriptionID",type="string",priority=1,JSONPath=".spec.subscriptionID"
+// +kubebuilder:printcolumn:name="Location",type="string",priority=1,JSONPath=".spec.location"
+// +kubebuilder:resource:path=azureclustertemplates,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+
+// AzureClusterTemplate is the Schema for the azureclustertemplates API
+type AzureClusterTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AzureClusterTemplateSpec `json:"spec,omitempty"`
```

Comment:
The spec should refer to a `template` which in turn will refer to `AzureClusterSpec`.  Something like we do for [AzureMachineTemplate](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d6a8184d4779bfddbc1efb87d0c910c55782e44a/api/v1alpha3/azuremachinetemplate_types.go#L32)

Reply:
got it :) Will make the changes now.